Framework/CurveFitting/src/Algorithms/SplineBackground.cpp

```diff
@@ -188,6 +172,39 @@ void SplineBackground::allocateBSplinePointers(int numBins, int ncoeffs) {
 }
 
 /**
+  * Calculates the bin weight from the error value passed
+  * logs a warning if the value is less than or equal to 0
+  * or a special floating point value
+  *
+  * @param errValue:: The error value to use within the calculation
+  *
+  * @return:: The bin weight to use within the GSL allocation
+  */
+double SplineBackground::calculateBinWeight(double errValue) {
+  double outBinWeight = 0;
+  if (errValue < 0 || !std::isnormal(errValue)) {
+    // Regardless of which warning we print we should
+    // set the bin weight
+    outBinWeight = 0;
```

Comment:
Yes but I wanted to make it explicit that we set it to 0. I'll initialize the value to -1 to catch if someone comes along and adds a branch or something that doesn't set the value.